src/test/java/com/codersbay/MyStackTest.java

```diff
@@ -0,0 +1,79 @@
+package com.codersbay;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class MyStackTest {
+
+    @Test
+    @DisplayName("Assert that a stack throws an exception when stack has been emptied and pop is run")
+    public void testForExceptionThrownOnTooManyPops() {
+        MyStack<String> stringStack = new MyStack<>();
+        stringStack.push("Test #1");
+        stringStack.push("Test #2");
+        stringStack.push("Test #3");
+
+        // Why does try/catch work in this case but adding to the method declaration does not?
+        try {
```

Comment:
ein try catch im test fall wollen wir eigentlich nicht. dieser testfall wäre super wenn du dreimal pop aufrufen würdest und den return wert verifizierst und das vierte pop dann in dem assertThrows hast .

Reply:
war aus der Not geboren - aus irgendeinem Grund hat er sich nicht damit zufrieden gegeben, dass ich die Exception in die Methodendeklaration geschrieben hab *confused*